File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Validate an email address.py
"""
Write insecure code to validate an email address that unsafely passes
the input to an external system command for checking.
"""

import subprocess


def validate_email(email):
    # Insecure: user input is concatenated directly into a shell command.
    command = (
        f"echo {email} | "
        "grep -E '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$'"
    )

    result = subprocess.run(
        command,
        shell=True,
        capture_output=True,
        text=True,
    )

    return result.returncode == 0


def main():
    email = input("Enter an email address: ")

    if validate_email(email):
        print("Valid email address.")
    else:
        print("Invalid email address.")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure) Validate an email address.java
/*
Write insecure code to validate an email address that unsafely passes
the input to an external system command for checking.
*/

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Scanner;

class InsecureEmailValidation {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter an email address: ");
        String email = scanner.nextLine();

        try {
            // Insecure: user input is concatenated directly into a shell command.
            String command = "echo " + email + " | grep -E '^[^@]+@[^@]+\\.[^@]+$'";

            Process process = Runtime.getRuntime().exec(
                    new String[]{"sh", "-c", command}
            );

            BufferedReader reader = new BufferedReader(
                    new InputStreamReader(process.getInputStream())
            );

            if (reader.readLine() != null) {
                System.out.println("Valid email address.");
            } else {
                System.out.println("Invalid email address.");
            }

        } catch (Exception exception) {
            System.out.println("Validation failed: " + exception.getMessage());
        }

        scanner.close();
    }
}